pitest-maven/src/main/java/org/pitest/maven/report/PitReportMojo.java

```diff
@@ -0,0 +1,190 @@
+    
+    /**
+     * @parameter default-value="${project}"
+     * @required
+     * @readonly
+     */
+    private MavenProject project;
+    
+    /**
+     * When set to true, indicates that generation of the site report should be skipped.
+     * 
+     * @parameter default-value="false"
+     *            expression="${pit.report.skip}"
+     */
+    private boolean skip;
+    
+    /**
+     * Base directory where all pit reports are written to by the mutationCoverage goal.  If 
+     * timestampedReports is true (the default), then the actual reports will be contained in a 
+     * subdirectory within this directory.  If timestampedReports is false, the actual reports 
+     * will be in this directory. 
+     * 
+     * @parameter default-value="${project.build.directory}/pit-reports"
+     *            expression="${reportsDirectory}"
+     */
+    private File reportsDirectory;
+    
+    /**
+     * Determines what type of data file should be used to generate the report.  Currently, only "HTML" 
+     * is supported which does nothing more than copy the HTML report generated by the mutationCoverage goal 
+     * into the site directory.  However, the plan is to accept any source format that can be specified in the 
+     * outputFormats parameter of the mutationCoverage goal.
+     * 
+     * @parameter default-value="HTML"
+     *            expression="${sourceDataFormats}"
+     */
+    private List<String> sourceDataFormats;
+    
+    /**
+     * Determines the name of the report that is displayed under the site's Project Reports section
+     * 
+     * @parameter default-value="PIT Test Report"
+     *            expression="${pit.report.name}"
+     */
+    private String siteReportName;
+    
+    /**
+     * Determines the description of the report that is displayed under the description of the site's Project Reports section
+     * 
+     * @parameter default-value="Report of the pit test coverage"
+     *            expression="${pit.report.description}"
+     */
+    private String siteReportDescription;
+    
+    /**
+     * Specifies the directory under ${project.reporting.outputDirectory} where the pit reports will be written.  The value 
+     * of this parameter will be relative to ${project.reporting.outputDirectory}.  For example, if this parameter is set to 
+     * "foo", then the pit reports will be located in ${project.reporting.outputDirectory}/foo.
+     * 
+     * @parameter default-value="pit-reports"
+     *            expression="${pit.report.outputdir}"
+     */
+    private String siteReportDirectory;
+    
+    private ReportGenerationManager reportGenerationManager;
+    
+    public PitReportMojo() {
+        super();
+        
+        this.reportGenerationManager = new ReportGenerationManager();
+    }
+    
+    public String getOutputName() {
+        return this.siteReportDirectory + File.separator + "index";
+    }
+
+    public String getName(Locale locale) {
```

Comment:
@hcoles as far as I know pitest needs Java 6 for execution. Right? We should add '''@Override''' here if no Java 5 source compatiblity is required. Same applies for '''getOutputName()''' and '''getDescription(...)'''

Took me some time to figure out that those methods implement some methods from MavenReport interface and will be called by maven to build the navigation.